agreement/service.go

```diff
@@ -117,9 +111,33 @@ func (s *Service) SetTracerFilename(filename string) {
 
 // Start executing the agreement protocol.
 func (s *Service) Start() {
```

Comment:
The changes to the agreement package are focused around the ability to start/stop the service repeatedly. I'm not sure how easy it would be to extract these, as there are other dependencies which would required to be updated along. ( i.e. network and node are two examples.. )